api/lib/opentelemetry/trace/samplers/decision.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/trace/samplers/basic_decision'
+
+module OpenTelemetry
+  module Trace
+    module Samplers
+      # The Decision class represents an arbitrary sampling decision. It can
+      # have a boolean value as a sampling decision and a collection of
+      # attributes to be attached to a sampled root span.
+      class Decision < BasicDecision
+        # Returns a new decision with the specified sampling decision and
+        # attributes.
+        #
+        # @param [Boolean] decision Whether or not a span should be sampled
+        # @param [Hash<String, Object>] attributes Attributes to be attached
+        #   to a root span
+        def initialize(decision:, attributes: nil)
+          super(decision: decision)
+          @attributes = attributes
```

Comment:
Is there a significant benefit to splitting the Decision functionality across `BasicDecision` and `Decision`? If we adopt a frozen constant empty Hash as the default attributes, there shouldn't be any performance difference. E.g.:
```ruby
class Decision
  EMPTY_ATTRIBUTES = {}.freeze

  def initialize(decision:, attributes: nil)
    @decision = decision
    @attributes = attributes || EMPTY_ATTRIBUTES
  end
end
```

Reply:
If attributes are immutable, there would not be a benefit to have multiple Decision classes. The only reason for this division is the uncertainty about the immutability of attributes.